modules/user/src/main/UserRepo.scala

```diff
@@ -181,6 +181,17 @@ object UserRepo {
       $set(F.profile -> Profile.profileBSONHandler.write(profile))
     ).void
 
+  def setUsernameCased(id: ID, username: String): Funit = {
+    if (id == username.toLowerCase) {
+      coll.update(
+        $id(id) ++ (F.changedCase $exists false),
+        $set(F.username -> username, F.changedCase -> true)
+      ) map { result =>
+          if (result.n == 0) fufail(s"You have already changed your username")
```

Comment:
I'm surprised this compiles. It makes the function evaluate to `Future[Future[Unit]]`.

`flatMap { result =>` should be used instead of `map`, in order to evaluate to `Future[Unit]` aka `Funit`.

Reply:
also there should be an `else funit` to signal success. I'm pretty sure the error is currently just discarded.